src/app/core/analytics/analytics.service.ts

```diff
@@ -53,16 +48,6 @@ export class AnalyticsService {
     });
   }
 
-  private subscribeToUserChanges() {
```

Comment:
very intentionally had the dependency direction this way round: In my opinion Session is a core service that should be as isolated as possible, while Analytics is a kind of "plugin" that can depend on all the services it requires but shouldn't be depended on, so that it can be swapped or turned on and off easily.

Reply:
In this case I think the same logic should then apply for the logging service for which the user is currently set at the same place where I added the analytics service now.
Also the analytics service became a dependency for all the sessions because I am tracking the `db.destroy` event with it that might be triggered in the LocalSession/DatabaseMigrationService.